eclair-core/src/main/scala/fr/acinq/eclair/wire/Onion.scala

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2019 ACINQ SAS
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package fr.acinq.eclair.wire
+
+import fr.acinq.bitcoin.ByteVector32
+import fr.acinq.eclair.ShortChannelId
+import fr.acinq.eclair.crypto.Sphinx
+import scodec.Codec
+import scodec.bits.ByteVector
+import scodec.codecs._
+
+/**
+  * Created by t-bast on 05/07/2019.
+  */
+
+case class OnionPacket(version: Int,
```

Comment:
To be consistent with how we map BOLT names to our own classes, `OnionPacket` should be `OnionRoutingPacket`, which may also help reduce the ambiguity described above between wire and Sphinx OnionPacket definitions